scalding-core/src/main/scala/com/twitter/scalding/Job.scala

```diff
@@ -498,6 +498,29 @@ class ScriptJob(cmds: Iterable[String]) extends Job(Args("")) {
   }
 }
 
+/**
+ * Allows custom counter verification logic when the job completes.
+ */
+trait CounterVerification extends Job {
+  /**
+   * Verify counter values. The job will fail if this returns false or throws an exception.
+   */
+  def verifyCounters(counters: Map[StatKey, Long]): Boolean
```

Comment:
We really try now to keep the type signature guiding how it is used. We don't want users throwing exceptions to the framework because these compose very poorly.

I think:
```
def verifyCounters(m: Map[StatKey, Long]) = Try {
  assert(isGood(m), s"Stats are incorrect, $m")
}
```
If we take the position that Try is too complex to use, we are kind of sunk since much of the API assumes understanding working with types like that.